### PR TITLE
PD-293: Adding a11y linter to leafyGreen

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

Bringing a11y linter into leafyGreen

To note: there are seven errors raised by this linter, but will address those in separate PRs and have disabled the lint rules in the interim. Doing this because I believe that there will be different levels of version-bumping the effected components, so did not want to batch all of these changes.

🎟 _Jira ticket:_ [PD-293](https://jira.mongodb.org/browse/PD-293)

## 🛠 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

~[ ] I have added tests that prove my fix is effective or that my feature works~
~[ ] I have added necessary documentation (if appropriate)~
~[ ] I have run `yarn changeset` and documented my changes~

### CODE REVIEW

File: packages/modal/src/Modal.tsx [hunk 1]
```diff
@@ -209,6 +210,7 @@ function Modal({
           >
             <div className={scrollContainer} ref={scrollContainerRef}>
               <div
+                // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
```
Comment (DesignerDave) [inline packages/modal/src/Modal.tsx]: Let's leave a "why" comment above this.

File: packages/modal/src/Modal.tsx [hunk 2]
```diff
@@ -200,6 +200,7 @@ function Modal({
     <Transition in={open} timeout={500} mountOnEnter unmountOnExit>
       {(state: string) => (
         <Portal>
+          {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,  jsx-a11y/no-static-element-interactions */}
```
Comment (DesignerDave) [inline packages/modal/src/Modal.tsx]: Can you explain what this is for / leave a comment in the file? I think as a general standard, any time we disable a linting rule, especially accessibility-related ones, we should have a comment explaining why.

File: packages/checkbox/src/Checkbox.tsx [hunk 3]
```diff
@@ -285,6 +285,7 @@ export default class Checkbox extends PureComponent<
         style={style}
         htmlFor={checkboxId}
       >
+        {/* eslint-disable-next-line jsx-a11y/no-redundant-roles */}
```
Comment (DesignerDave) [inline packages/checkbox/src/Checkbox.tsx]: See my comment below.

File: packages/modal/src/Modal.tsx (hunk 2 above)
Comment (brookescarlett) [reply packages/modal/src/Modal.tsx]: Was only disabling for now, and was planning on making separate PRs to each component to make them comply with new linter rules. Just wanted to break it up, so it wasn't one large PR. Let me know if you'd prefer if I just make the updates right now, and can just include multiple changesets in this PR.

File: packages/modal/src/Modal.tsx [hunk 4]
```diff
@@ -202,14 +202,17 @@ function Modal({
         <Portal>
           <div
             {...rest}
+            role="none"
```
Comment (DesignerDave) [inline packages/modal/src/Modal.tsx]: What's the difference between this, and not setting `role` at all?

File: packages/modal/src/Modal.tsx [hunk 5]
```diff
@@ -226,7 +229,7 @@ function Modal({
                   onClick={handleClose}
                   className={closeButton}
                   data-dismiss="modal"
-                  aria-hidden="true"
+                  tabIndex={0}
```
Comment (DesignerDave) [inline packages/modal/src/Modal.tsx]: We seem to be setting tabIndex twice?

Comment (brookescarlett) [reply packages/modal/src/Modal.tsx]: -1 on the container, 0 on the "x" close button

File: packages/modal/src/Modal.tsx (hunk 4 above)
Comment (brookescarlett) [reply packages/modal/src/Modal.tsx]: Added a comment!

**Pull request merged**